# Configuration for processing ROS2 bag files
# This file specifies the paths, storage format, and topics to extract from the ROS bag.

# Path to the input ROS2 bag file
bag_path: "/path/to/your.bag"

# Directory where the exported data will be saved
output_dir: "/path/to/output_directory"

# Storage format for the exported data. Options are "sqlite3" or "mcap"
storage_id: "mcap"

# List of topics to extract from the ROS2 bag
topics:
  # The timestamp of the first topic in this list will be used as the reference for sensor synchronisation
  # Configuration for the point cloud topic. 
  - name: "/pointcloud_topic"   # Name of the topic to extract
    type: "PointCloud2"         # Data type of the topic
    sample_interval: 10         # Write one sample every 10 messages
    topic_dir: "lidar"          # Output data in <output_dir>/<bag_name>/lidar

  # Configuration for the RGB image topic
  - name: "/rgb_image_topic"    # Name of the topic to extract
    type: "Image"               # Data type of the topic
    encoding: "rgb8"            # Encoding format for the image
    sample_interval: 5          # Write one sample every 5 messages
    topic_dir: "cam/raw"        # Output data in <output_dir>/<bag_name>/cam/raw

  # Configuration for the compressed image topic
  - name: "/compressed_image_topic"  # Name of the topic to extract
    type: "CompressedImage"          # Data type of the topic
    sample_interval: 5               # Write one sample every 5 messages
    topic_dir: "cam/compressed"      # Output data in <output_dir>/<bag_name>/cam/compressed

  # Configuration for the IMU (Inertial Measurement Unit) topic
  - name: "/imu_topic"          # Name of the topic to extract
    type: "IMU"                 # Data type of the topic
    sample_interval: 20         # Write one sample every 20 messages
    topic_dir: "imu"            # Output data in <output_dir>/<bag_name>/imu

  # Configuration for the GPS topic
  - name: "/gps_topic"          # Name of the topic to extract
    type: "GPS"                 # Data type of the topic
    sample_interval: 1          # Write every message
    topic_dir: "gps"            # Output data in <output_dir>/<bag_name>/gps
